Sources/JWTKit/EdDSA/EdDSAKey.swift

```diff
@@ -0,0 +1,51 @@
+import Foundation
+import Crypto
+
+public struct EdDSAKey {
+            
+    let keyPair: OctetKeyPair
+    var publicKey: Data? {
+        keyPair.publicKey
+    }
+    var privateKey: Data? {
+        keyPair.privateKey
+    }
+    let curve: JWK.Curve
+    
+    public init(x: String?, d: String? = nil, curve: JWK.Curve = .ed25519) throws {
+        try self.init(
+            publicKey: x.flatMap { $0.data(using: .utf8) }.map { Data($0.base64URLDecodedBytes()) },
+            privateKey: d.flatMap { $0.data(using: .utf8) }.map { Data($0.base64URLDecodedBytes()) },
+            curve: curve
+        )
+    }
+    
+    public init(publicKey: Data? = nil, privateKey: Data? = nil, curve: JWK.Curve = .ed25519) throws {
```

Comment:
I'm a bit suspicious as to why we're letting users tell us about the curve. I suppose it's to let us do `ed448` in the future, but the fact that the API invites you to pass `.p256` to it is a bit weird.

Reply:
I agree. That's weird on the level of specific keys. I scoped the supported curves for the particular key and added a mapping from the generic JWK definition, which could always be mismatched when parsed from JSON. c34e92a69dad2d4fcffcf51023a4e44f360cb179